src/factorio_randovania_mod/lua_src/control.lua

```diff
@@ -153,43 +204,57 @@ script.on_event(defines.events.on_player_main_inventory_changed, on_player_main_
 
 ---@param event EventData.on_player_joined_game
 local function on_player_joined_game(event)
-    -- We shouldn't need this.
-    -- if not storage.player_pending_freebies then
-    --     storage.player_pending_freebies = {}
-    --     storage.total_freebies = {}
-    -- end
-
     if not storage.player_pending_freebies[event.player_index] then
         -- New joiner!
         local freebies = {}
-        for _, freebie in pairs(storage.total_freebies) do
-            table.insert(freebies, freebie)
+
+        if not strict_multiplayer_freebie then
+            for freebie, amount in pairs(storage.items_with_freebies) do
+                if amount == 0 then
+                    freebies[freebie] = 1
+                end
+            end
         end
         storage.player_pending_freebies[event.player_index] = freebies
     end
+    distribute_freebies()
     give_pending_freebies(event.player_index)
 end
 
-
 script.on_event(defines.events.on_player_joined_game, on_player_joined_game)
 
+
+---@param event EventData.on_player_toggled_map_editor
+local function on_player_toggled_map_editor(event)
```

Comment:
toggled map editor?

Reply:
If you use `/editor`, unlock some tech, then `/editor` to leave, it'll run the check. Relevant because you don't have a character while in editor mode so you don't get your freebies.

Yes, the main use case is helping me dev lol